src/OpenLoco/src/S5/S5.cpp

```diff
@@ -109,7 +113,71 @@ namespace OpenLoco::S5
         return result;
     }
 
-    static void drawPreviewImage(void* pixels, Ui::Size size)
+    // 0x0046DB4C
+    void drawScenarioPreviewImage()
+    {
+        auto& options = S5::getOptions();
+        const auto kPreviewSize = sizeof(options.preview[0]);
+        const auto kMapSkipFactor = kMapRows / kPreviewSize;
+
+        for (auto y = 0U; y < kPreviewSize; y++)
+        {
+            for (auto x = 0U; x < kPreviewSize; x++)
+            {
+                uint8_t colour = 0;
+                auto pos = TilePos2(kMapColumns - (x + 1) * kMapSkipFactor + 1, y * kMapSkipFactor + 1);
```

Comment:
This is some deep nesting can we split it off into a function

Reply:
Could move the tile colour stuff into a function of its own, I suppose. Something like `getPreviewColourByTilePos`?